src/types/resolveSignatures.ts

```diff
@@ -140,6 +140,11 @@ export function resolveSignatures(ctx: CompilerContext) {
                         `Unsupported map format: ${src.type.format}`,
                     );
                 }
+                if (src.type.keyFormat === "coins") {
+                    throwCompilationError(
+                        `Unsupported format ${src.type.keyFormat} for map key`,
```

Comment:
Hmmm, why cannot we support the `coins` storage annotation?

Reply:
AFAIR, there is support in TVM for variable size keys in hashmaps

Reply:
There are no TVM instructions for VarHashmap at the moment

Reply:
Ah, then perhaps `tvm.pdf` was a bit too optimistic :)